webui/react/src/pages/TrialDetails/TrialInfoBox.test.tsx

```diff
@@ -0,0 +1,231 @@
+  autoRestarts: 0,
+  bestAvailableCheckpoint: {
+    endTime: '2022-08-05T05:56:19.781968Z',
+    resources: {
+      'code/': 0,
+      'code/adaptive.yaml': 418,
+      'code/const.yaml': 349,
+      'code/data.py': 3254,
+      'code/mnist.py': 2996,
+      'code/model_def.py': 1445,
+      'code/README.md': 1492,
+      'code/startup-hook.sh': 59,
+      'load_data.json': 2664,
+      'state_dict.pth': 680105,
+      'workload_sequencer.pkl': 89,
+    },
+    state: 'COMPLETED' as const,
+    totalBatches: 58,
+    uuid: '3b780651-0250-4307-8ec0-fac09a688970',
+  },
+  bestValidationMetric: {
+    endTime: '2022-08-05T05:56:21.150552Z',
+    metrics: { accuracy: 0.1809730976819992, val_loss: 2.1605682373046875 },
+    totalBatches: 58,
+  },
+  endTime: '2022-08-05T05:56:49.524883Z',
+  experimentId: 1,
+  hyperparameters: {
+    global_batch_size: 64,
+    hidden_size: 64,
+    learning_rate: 0.0951845051299588,
+  },
+  id: 2,
+  latestValidationMetric: {
+    endTime: '2022-08-05T05:56:21.150552Z',
+    metrics: { accuracy: 0.1809730976819992, val_loss: 2.1605682373046875 },
+    totalBatches: 58,
+  },
+  logRetentionDays: 12,
+  startTime: '2022-08-05T05:55:55.152888Z',
+  state: 'COMPLETED' as const,
+  totalBatchesProcessed: 58,
+  totalCheckpointSize: 58,
+};
+
+const mockTrial3: TrialDetails = {
+  autoRestarts: 0,
+  bestAvailableCheckpoint: {
+    endTime: '2022-08-05T05:56:41.515152Z',
+    resources: {
+      'code/': 0,
+      'code/adaptive.yaml': 418,
+      'code/const.yaml': 349,
+      'code/data.py': 3254,
+      'code/mnist.py': 2996,
+      'code/model_def.py': 1445,
+      'code/README.md': 1492,
+      'code/startup-hook.sh': 59,
+      'load_data.json': 2665,
+      'state_dict.pth': 680105,
+      'workload_sequencer.pkl': 89,
+    },
+    state: 'COMPLETED' as const,
+    totalBatches: 58,
+    uuid: 'b3d4ed7e-625c-4083-9e6c-ffba4ac03729',
+  },
+  bestValidationMetric: {
+    endTime: '2022-08-05T05:56:42.958782Z',
+    metrics: { accuracy: 0.3433544337749481, val_loss: 1.7270305156707764 },
+    totalBatches: 58,
+  },
+  endTime: '2022-08-05T05:56:49.345991Z',
+  experimentId: 1249,
+  hyperparameters: {
+    global_batch_size: 64,
+    hidden_size: 64,
+    learning_rate: 0.06607116139078467,
+  },
+  id: 3567,
+  latestValidationMetric: {
+    endTime: '2022-08-05T05:56:42.958782Z',
+    metrics: { accuracy: 0.3433544337749481, val_loss: 1.7270305156707764 },
+    totalBatches: 58,
+  },
+  logRetentionDays: -1,
+  startTime: '2022-08-05T05:56:16.181755Z',
+  state: 'COMPLETED' as const,
+  totalBatchesProcessed: 58,
+  totalCheckpointSize: 58,
+};
+
+describe('Trial Info Box', () => {
+  describe('should show log retention days box', () => {
+    it('should show Log Retention box with "-" for days', async () => {
+      setup(mockTrial1, mockExperiment);
+      expect(await screen.findByText('Log Retention Days')).toBeVisible();
+      expect(await screen.findByText('-')).toBeVisible();
```

Comment:
can we narrow the `-` element using a query because `-` can be somewhere else? same for other test cases